src/App/Service/ExcelImport.php

```diff
@@ -129,16 +130,45 @@ private function saveDoc(Worksheet $sheet): LsDoc
                 )
             )
         );
-        $doc->setLicence($this->getCellValueOrNull($sheet, 14, 2));
-        // col 14 - Licence title
-        // col 15 - Licence text
-        $doc->setNote($this->getCellValueOrNull($sheet, 16, 2));
+
+        $licence = $this->getLicence($sheet);
+
+        $doc->setLicence($licence);
+
+        $doc->setNote($this->getCellValueOrNull($sheet, 17, 2));
 
         $this->getEntityManager()->persist($doc);
 
         return $doc;
     }
 
+    private function getLicence(Worksheet $sheet): LsDefLicence
+    {
+        $uri = $this->getCellValueOrNull($sheet, 14, 2);
+        $title = $this->getCellValueOrNull($sheet, 15, 2);
+        $licenceText = $this->getCellValueOrNull($sheet, 16, 2);
+
+        // Look for this licence locally
+        $licence = $this->getEntityManager()->getRepository(LsDefLicence::class)
+            ->findOneByUri($uri);
```

Comment:
Should probably check if `$uri` is empty before trying to find a licence that matches (or trying to create one), as it is possible for there to be no licence.

Reply:
sure thing.

Reply:
It would be better to check if `$uri` is empty before using it to query the database, but it will work.